.github/workflows/pr-create-release-build.yml

```diff
@@ -0,0 +1,207 @@
+        run: |
+          git add config/overlays/odh/params.env
+          git commit -m "Update image refs for odh release"
+          git push origin $BRANCH_NAME
+      - name: Create Pull Request
+        id: create-pull-request
+        run: | 
+          PR_URL=$(gh pr create -B ${{ github.ref }} -H ${{ env.BRANCH_NAME }} --title '[ODH Release] Update images for ${{ github.event.inputs.tag_name }}' --body 'Update images for ${{ github.event.inputs.tag_name }}')
+          echo "PR_URL=${PR_URL}" >> $GITHUB_ENV
+          pr_number=$(echo "$PR_URL" | grep -o -E '[0-9]+$')
+          echo "pr_number=${pr_number}" >> $GITHUB_OUTPUT
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+  wait-checks:
+    runs-on: ubuntu-latest
+    needs: [ check-prev-tag,create-pr ] 
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Watching PR if Checks finished without errors
+        id: wait-checks
+        run:
+          gh pr checks  ${{ needs.create-pr.outputs.pr_number }}  --watch --fail-fast 
+        env:     
+         GH_TOKEN: ${{ github.token }}
+  comment-lgtm:
+    needs: [ check-prev-tag,create-pr,wait-checks ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+      
+      - name: Add comment to trigger lgtm label
+        if: ${{ needs.wait-checks.result == 'success' }}
+        run: |
+          gh pr comment ${{ needs.create-pr.outputs.pr_number }} --body "/lgtm"
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+  rename-image:
+    needs: [ check-prev-tag,create-pr,wait-checks,comment-lgtm]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to Quay.io
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.Q_PASSWORD }}
+    
+      - name: Wait for PR to be merged
+        uses: nick-fields/retry@v3
+        with:
+          timeout_minutes: 60
+          max_attempts: 60
+          retry_wait_seconds: 120
+          shell: bash
+          command: |
+            PR_STATUS=$(gh pr view ${{ needs.create-pr.outputs.pr_number }} --json state --jq '.state')
+            echo "PR Status: $PR_STATUS"
+            if [ "$PR_STATUS" != "MERGED" ]; then
+              echo "PR is not yet merged. Retrying..."
+              exit 1
+            fi
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: pull image (with retry)
+        uses: nick-fields/retry@v3
+        with:
+          timeout_minutes: 2
+          max_attempts: 3
+          retry_wait_seconds: 120
+          shell: bash
+          command: docker pull quay.io/${{ vars.QUAY_OWNER }}/kserve-controller:pr-${{ needs.create-pr.outputs.pr_number }}

```

Comment:
You only re-tag kserve-controller, but you need to re-tag all 4 containers: kserve-controller, kserve-storage-initializer, kserve-agent and kserve-router.